apps/multicountdown/multicountdown.star

```diff
@@ -0,0 +1,174 @@
+                for x in range(int(eventCount)):
+                    if config.str("event_" + str(x) + "_time"):
+                        eventTime = time.parse_time(config.str("event_" + str(x) + "_time")).in_location(timezone)
+                        dateDiff = eventTime - time.now().in_location(timezone)
+                        eventArray.append(dateDiff)
+                    else:
+                        break
+
+            #Let's find the closest  event:
+            closestEventIndex = find_shortest_non_negative_time_index(eventArray)
+            if closestEventIndex > -1:
+                parsedTime = time.parse_duration(eventArray[closestEventIndex])
+                days = int(parsedTime.hours / 24)
+                hours = int(parsedTime.hours)
+                minutes = int(parsedTime.minutes)
+                if hours < 1 and minutes < 16 and minutes > 5:
+                    color = "#FFFF00"
+                if hours < 1 and minutes < 6:
+                    color = "#FF0000"
+                formatted_duration = format_duration(days, hours, minutes)
+                output = formatted_duration + " until " + config.str("event" + str(closestEventIndex))
+            else:
+                output = "NO FUTURE EVENTS"
+        else:
+            output = "NO EVENTS"
+    else:
+        output = "WAITING FOR DATA"
+
+    #Let's draw:
+    return render.Root(
+        delay = 500,
+        child = render.Column(
+            expanded = True,
+            main_align = "center",
+            cross_align = "center",
+            children = [
+                render.Row(
+                    expanded = True,
+                    main_align = "center",
+                    cross_align = "center",
+                    children = [
+                        render.Animation(
+                            children = [
+                                render.Text(
+                                    content = now.format("3:04 PM"),
+                                    font = "6x13",
+                                ),
+                                render.Text(
+                                    content = now.format("3 04 PM"),
+                                    font = "6x13",
+                                ),
+                            ],
+                        ),
+                    ],
+                ),
+                render.Row(
+                    expanded = True,
+                    main_align = "center",
+                    cross_align = "center",
+                    children = [
+                        render.WrappedText(
+                            content = str(output),
+                            color = color,
+                            font = "CG-pixel-3x5-mono",
+                            align = "center",
+                        ),
+                    ],
+                ),
+            ],
+        ),
+    )
+
+def more_options(eventCount):
+    returnArray = []
+    if eventCount > "0":
```

Comment:
Would you mind putting an upper limit on eventCount? Just someone entering a very large number doesn't cause weird failures that are confusing to users. I'm thinking 20 would be more than enough?

Reply:
No problem, done.